src/search_algo.hpp

```diff
@@ -598,7 +598,13 @@ _setFrames(TBlastMatch                          & bm,
         bm.qFrameShift = (m.qryId % 3) + 1;
         if (m.qryId % 6 > 2)
             bm.qFrameShift = -bm.qFrameShift;
-    } else if constexpr (seqan::qHasRevComp(TLocalHolder::TGlobalHolder::blastProgram))
+    } else if constexpr (TLocalHolder::TGlobalHolder::c_redAlph == AlphabetEnum::DNA3BS)
+    {
+        bm.qFrameShift = 1;
+        if (m.qryId % 4 > 1)
+            bm.qFrameShift = -bm.qFrameShift;
```

Comment:
```suggestion
        bm.qFrameShift = (m.qryId % 4 > 1) ? -1 : 1;
```

Should be easier to optimise.

Reply:
Anyway, this means

0 -> frameshift 1
1 -> frameshift 1
2 -> frameshift -1
3 -> frameshift -1

Reply:
So frameshift actually implies "reverse comlpement". The information for "second translation frame" is lost. Or is it the other way around?

Reply:
Yes, I thought this is what the frameshift implies for the normal nucleotide mode as well. Please correct me if that was wrong.